src/tensor_ops/utilities/compatibility.cuh

```diff
@@ -0,0 +1,54 @@
+#include "cuda_fp16.h"
+
+// Table showing which features are supported on which compute capability
+// https://docs.nvidia.com/cuda/cuda-c-programming-guide/#features-and-technical-specifications
+
+// FIXME: the minimum compute capabilities are just guesses since the table is not specific enough
+
+#if __CUDA_ARCH__ < 700
+__device__ __forceinline__ __half __hmax_nan(__half a, __half b) {
+    return __hisnan(a) ? a : (__hisnan(b) ? b : __hmax(a, b));
+}
+__device__ __forceinline__ __half __hmin_nan(__half a, __half b) {
+    return __hisnan(a) ? a : (__hisnan(b) ? b : __hmin(a, b));
+}
+#endif
+
+
+// https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#atomicadd
+// > The 16-bit __half floating-point version of atomicAdd() is only supported by devices of compute capability 7.x and higher.
+// We assume that atomicCAS for shorts has the same compute capability restrictions.
+#if __CUDA_ARCH__ < 700
+
+// Inspired by https://github.com/torch/cutorch/blob/master/lib/THC/THCAtomics.cuh#L96-L119
```

Comment:
It seems like this is doing atomicAdd with `unsigned int` instead of `unsigned short`? Should we be doing that as well? Then we could basically just copy the function from the link without having to redefine atomicCAS